### PR TITLE
Update to OTLP proto 0.12.0

### PR DESCRIPTION
**Pull request by tigrannajaryan**
Resolves https://github.com/open-telemetry/opentelemetry-collector/issues/4717

- Removed all deprecated metric messages and corresponding conversion code and tests.
- Removed deprecated_code from span status and corresponding conversions and tests.
- Renamed logs field to log_records in InstrumentationLibraryLogs message.

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
@open-telemetry/collector-approvers please review the deletions of metrics code carefully. I want to make sure I didn't break anything and didn't delete any code that was supposed to stay.

**Review by Aneurysm9 — APPROVED** [review]
It gets a bit hard to follow the changes in the generated `*.pb.go` files, but it looked like the metrics changes were what I'd expect.  The deprecated types were removed and from there it looked like noise from shuffling things around.

**Review by codeboten — COMMENTED** [review]
I think the changes all look good, I'd like to merge this after this next release though

File: model/pdata/logs.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ func (ld Logs) LogRecordCount() int {
 		ill := rs.InstrumentationLibraryLogs()
 		for i := 0; i < ill.Len(); i++ {
 			logs := ill.At(i)
-			logCount += logs.Logs().Len()
+			logCount += logs.LogRecords().Len()
```
Comment (bogdandrutu) [inline model/pdata/logs.go:88]: Add this code please, to help with migration:
```golang
// Deprecated: use `LogRecords`.
func (ms InstrumentationLibraryLogs) Logs() LogSlice {
	return ms.LogRecords()
}
```

File: model/pdata/generated_log.go [hunk 2]
```diff
@@ -401,16 +401,16 @@ func (ms InstrumentationLibraryLogs) SetSchemaUrl(v string) {
 	(*ms.orig).SchemaUrl = v
 }
 
-// Logs returns the Logs associated with this InstrumentationLibraryLogs.
-func (ms InstrumentationLibraryLogs) Logs() LogSlice {
-	return newLogSlice(&(*ms.orig).Logs)
+// LogRecords returns the LogRecords associated with this InstrumentationLibraryLogs.
+func (ms InstrumentationLibraryLogs) LogRecords() LogSlice {
```
Comment (bogdandrutu) [inline model/pdata/generated_log.go:405]: Would be good to rename `LogSlice` to `LogRecordSlice`. If you do that add in pdata/logs.go:

```golang
// Deprecated: use LogRecordSlice
type LogSlice = LogRecordSlice
```

**Review by bogdandrutu — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
> I think the changes all look good, I'd like to merge this after this next release though

Yes, let's hold this until the release is done.

**Review by codeboten — APPROVED** [review]
